### PR TITLE
Multiple rabbitmq bug fixes

### PR DESCRIPTION
**Pull request by schandrika**
1. Minor change to rabbitmq dependency install script.  Updated readme
2. Fix for #1924 
3. Fix for #1925 
4. Fix for #1927 
5. Fix for #1928 
6. Fix for #1920

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,48 +56,27 @@ http://ryanstutorials.net/linuxtutorial/
...
+
+  If you are running an Debian or CentOS system, you can install the RabbitMQ dependencies by running the rabbit 
+  dependencies script, passing in the os name and approriate distribution as a parameter. The following are  
+  supported
+
+    debian bionic (for Ubuntu 18.04)
+    denian xenial (for Ubuntu 16.04)
+    debian xenial (for Linux Mint 18.04)
+    debian stretch (for Debian Stretch)
+    debian jessie (for Debian Jessie)
```
Comment (craig8) [inline README.md]: Jessie is not supported any longer due to remove of depencency from apt repo for jessie

File: README.md [hunk 2]
```diff
@@ -56,48 +56,27 @@ http://ryanstutorials.net/linuxtutorial/
...
+    centos 7 (for CentOS 7)
+    centos 6 (for CentOS 6)
   
+  Example command
   ``` 
-  ~/volttron/scripts/rabbit_dependencies.sh xenial
+  ~/volttron/scripts/rabbit_dependencies.sh denian xenial
```
Comment (craig8) [inline README.md]: denian should be debian

File: bootstrap.py [hunk 3]
```diff
@@ -276,6 +277,15 @@ def update(operation, verbose=None, upgrade=False, offline=False,
 
 
 def install_rabbit(rmq_install_dir):
+
+    # try:
+    process = subprocess.Popen(["which", "erl"], stderr=subprocess.PIPE,  stdout=subprocess.PIPE)
+    (output, error) = process.communicate()
+    if process.returncode != 0:
+        sys.stderr.write("ERROR:\n Unable to find erlang in path. Please install necessary pre-requisites. "
+                         "Reference: https://github.com/schandrika/volttron/blob/rabbitmq-volttron/README.md")
```
Comment (craig8) [inline bootstrap.py]: Should this be your repo?

File: scripts/rabbit_dependencies.sh [hunk 4]
```diff
@@ -14,8 +14,8 @@ function exit_on_error {
 function print_usage {
  echo "
 Command Usage:
-<path>/rabbit_dependencies.sh <ubuntu or centos> <distribution name or centos version>
-Valid Ubuntu distributions: ${list[@]}
+<path>/rabbit_dependencies.sh <debian or centos> <distribution name or centos version>
+Valid Debian distributions: ${list[@]}
```
Comment (craig8) [inline scripts/rabbit_dependencies.sh]: should be debian based....ubuntu is debian based...

File: scripts/rabbit_dependencies.sh [hunk 5]
```diff
@@ -93,13 +93,12 @@ fi
 
 $prefix pwd > /dev/null
 
-if [ "$os_name" == "ubuntu" ]; then
+if [ "$os_name" == "debian" ]; then
     install_on_ubuntu
```
Comment (craig8) [inline scripts/rabbit_dependencies.sh]: if debian install_on_ubuntu?

File: volttron/platform/instance_setup.py [hunk 6]
```diff
@@ -76,7 +76,8 @@ def _load_config():
             config_opts[option] = parser.get('volttron', option)
 
 
-def _update_config_file(instance_name='volttron1'):
+def _update_config_file(instance_name=None):
+    _load_config()
```
Comment (craig8) [inline volttron/platform/instance_setup.py]: note get_home is already called within load_config as well as the path exists etc.

File: README.md (hunk 1 above)
Comment (schandrika) [reply README.md]: Removed Jessie

File: README.md (hunk 2 above)
Comment (schandrika) [reply README.md]: Fixed

File: bootstrap.py (hunk 3 above)
Comment (schandrika) [reply bootstrap.py]: Fixed

File: scripts/rabbit_dependencies.sh (hunk 4 above)
Comment (schandrika) [reply scripts/rabbit_dependencies.sh]: Fixed

File: scripts/rabbit_dependencies.sh (hunk 5 above)
Comment (schandrika) [reply scripts/rabbit_dependencies.sh]: fixed

File: volttron/platform/instance_setup.py (hunk 6 above)
Comment (schandrika) [reply volttron/platform/instance_setup.py]: get_home needs to be called in update method too so that it can create new config if one doesn't exist

**Comment by craig8** [comment]
````
volttron)osboxes@v2 ~/repos/volttron-rabbitmq $ vcfg --rabbitmq single examples/configurations/rabbitmq/rabbitmq_config_local_ssl.yml 

Your VOLTTRON_HOME currently set to: /home/osboxes/.try-vhome-pr

Is this the volttron you are attempting to setup?  [Y]: y
Traceback (most recent call last):
  File "/home/osboxes/repos/volttron-rabbitmq/env/bin/vcfg", line 11, in <module>
    load_entry_point('volttron', 'console_scripts', 'vcfg')()
  File "/home/osboxes/repos/volttron-rabbitmq/volttron/platform/config.py", line 593, in _main
    main()
  File "/home/osboxes/repos/volttron-rabbitmq/volttron/platform/instance_setup.py", line 613, in main
    process_rmq_inputs(args.rabbitmq, args.instance_name)
  File "/home/osboxes/repos/volttron-rabbitmq/volttron/platform/instance_setup.py", line 553, in process_rmq_inputs
    copy(args[1], vhome_config)
  File "/usr/lib/python2.7/shutil.py", line 119, in copy
    copyfile(src, dst)
  File "/usr/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 2] No such file or directory: '/home/osboxes/.try-vhome-pr/rabbitmq_config.yml'
````

**Comment by shwethanidd** [comment]
In readme, can you add - ~/volttron/scripts/rabbit_dependencies.sh <os> <distribution>. Currently the command shows that it should take single parameter.

**Comment by schandrika** [comment]
I see
 Example command
  ``` 
  ~/volttron/scripts/rabbit_dependencies.sh debian xenial
  ```
In readme. Are you referring to a different place ?

**Comment by craig8** [comment]
I am good with this

@shwethanidd merge if you don't have any objections

**Pull request merged**